From 7c00086ee906e5be5c373ee30a6ba031e2e076c5 Mon Sep 17 00:00:00 2001
From: Jinping Wang <jinping.wang@amlogic.com>
Date: Thu, 1 Nov 2018 11:29:39 +0800
Subject: [PATCH] cec: cec-framework [1/1]

PD# SWPL-887

Problem:
when chang to playback input may not be controlled

Solution:
send Active Source when receive Routing Change for playback

Verify:
test ok using r311

Change-Id: Id95ddbbb668cb809ad01017d5ef76ca459d359cb
Signed-off-by: Jinping Wang <jinping.wang@amlogic.com>
---
 .../core/java/com/android/server/hdmi/HdmiCecLocalDevicePlayback.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/services/core/java/com/android/server/hdmi/HdmiCecLocalDevicePlayback.java b/services/core/java/com/android/server/hdmi/HdmiCecLocalDevicePlayback.java
index b3aef04..49345e5 100644
--- a/services/core/java/com/android/server/hdmi/HdmiCecLocalDevicePlayback.java
+++ b/services/core/java/com/android/server/hdmi/HdmiCecLocalDevicePlayback.java
@@ -286,6 +286,10 @@ final class HdmiCecLocalDevicePlayback extends HdmiCecLocalDevice {
         assertRunOnServiceThread();
         int newPath = HdmiUtils.twoBytesToInt(message.getParams(), 2);
         maySetActiveSource(newPath);
+        int port = getLocalPortFromPhysicalAddress(newPath);
+        if (port == 0) {
+            maySendActiveSource(message.getSource());
+        }
         return true;  // Broadcast message.
     }
 
-- 
2.10.2